SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCloudLauncherSettings.cpp

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#include "SpatialGDKEditorCloudLauncherSettings.h"
+
+#include "Dom/JsonObject.h"
+#include "Internationalization/Regex.h"
+#include "Misc/FileHelper.h"
+#include "Serialization/JsonReader.h"
+#include "Serialization/JsonSerializer.h"
+#include "Settings/LevelEditorPlaySettings.h"
+#include "SpatialGDKEditorSettings.h";
+#include "Templates/SharedPointer.h"
+
+USpatialGDKEditorCloudLauncherSettings::USpatialGDKEditorCloudLauncherSettings(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	const USpatialGDKEditorSettings * SpatialEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	SnapshotPath.FilePath = FPaths::Combine(SpatialEditorSettings->GetSpatialOSSnapshotFolderPath(), SpatialEditorSettings->GetSpatialOSSnapshotFile());
+	ProjectName = GetProjectNameFromSpatial();
+}
+
+FString USpatialGDKEditorCloudLauncherSettings::ToString()
+{
+	TArray<FStringFormatArg> Args;
+
+	Args.Add(ProjectName);
+	Args.Add(PrimaryDeploymentName);
+	Args.Add(AssemblyName);
+	Args.Add(PrimaryLaunchConfigPath.FilePath);
+	Args.Add(SnapshotPath.FilePath);
+	Args.Add(SimulatedPlayerDeploymentName);
+	Args.Add(SimulatedPlayerLaunchConfigPath.FilePath);
+	Args.Add(NumberOfSimulatedPlayers);
+	Args.Add(bPrimaryDeploymentNameIsValid);
+	Args.Add(bAssemblyNameIsValid);
+	Args.Add(bProjectNameIsValid);
+	Args.Add(SimulatedPlayersIsEnabled);
+
+	return FString::Format(TEXT(
+		"ProjectName={0}, "
+		"PrimaryDeploymentName={1}"
+		"AssemblyName={2}"
+		"PrimaryLaunchConfigPath={3}, "
+		"SnapshotPath={4}, "
+		"SimulatedPlayerDeploymentName={5}, "
+		"SimulatedPlayerLaunchConfigPath={6}, "
+		"NumberOfSimulatedPlayers={7}, "
+		"bPrimaryDeploymentNameIsValid={8}, "
+		"bAssemblyNameIsValid={9}, "
+		"bProjectNameIsValid={10}, "
+		"SimulatedPlayersIsEnabled={11}")
+		, Args);
+}
+
+FString USpatialGDKEditorCloudLauncherSettings::GetProjectNameFromSpatial()
```

Comment:
Could we make this const member function?

Reply:
Definitely 👍.